plugins/ubiquiti/unifi_api

```diff
@@ -0,0 +1,1138 @@
+    # Show detailed graphs for each device
+
+    env.enable_xfer_network yes
+    # Show transfer statistics per named network
+    env.enable_detail_xfer_network yes
+    # Show detailed graphs for each named network
+
+    env.enable_xfer_radio yes
+    # Show transfer statistics per radio
+    env.enable_detail_xfer_radio yes
+    # Show detailed graphs for each radio
+
+
+=head1 CAPABILITIES
+
+This plugin requires multigraph.
+
+This plugin supports L<DIRTYCONFIG|http://guide.munin-monitoring.org/en/latest/plugin/protocol-dirtyconfig.html>
+
+=head1 DEPENDENCIES
+
+=over
+
+=item WWW::Curl::Easy
+
+Perl extension interface for libcurl
+
+=item JSON
+
+ JSON (JavaScript Object Notation) encoder/decoder
+
+=back
+
+=head1 PERFORMANCE CONCERNS
+
+The main performance concern on this is the huge number of graphs that may be
+generated.  Using the cron version of munin-graph may hurt a lot.
+
+A bit of a case study:
+
+                 | My Site  | UBNT Demo
+---------------------------------------
+Devices          |       8  |     126
+AP's             |       4  |     118
+24xSwitch        |       1  |       5
+8xSwitch         |       2  |       2
+Output Bytes     |  64,262  | 431,434
+Output Lines     |   1,761  |  14,586
+Output Graphs    |      77  |     530
+
+So, just note that the growth in the amount of graphed date can be extream.
+
+
+=head1 LICENSE
+
+Copyright (C) 2018 J.T.Sage (jtsage@gmail.com)
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see L<http://www.gnu.org/licenses/>.
+
+=head1 VERSION
+
+Most recent version available from 
+
+L<https://github.com/jtsage/unifi-munin-plugins>
+
+=head1 MAGICK MARKERS
+
+  #%# family=manual
+  #%# capabilities=autoconf
+
+=cut
+
+use warnings;
+use strict;
+use utf8;
+
+# Check dependancies
+my @errorCode;
+my $me = (split '/', $0)[-1];
+
+if (! eval {require JSON; JSON->import(); 1; } ) {
+	push @errorCode, "JSON module not found";
+}
+if (! eval {require WWW::Curl::Easy; 1;} ) {
+	push @errorCode, "WWW::Curl::Easy module not found";
+}
+
+# Provide autoconf info
+if ( defined($ARGV[0]) && $ARGV[0] eq "autoconf" ) {
+	push @errorCode, "Requires configuration";
+	print "no (", join(", " => @errorCode), ")\n";
+	exit(0);
+}
+
+# Fail on not found dependancies
+if ( @errorCode != 0 ) {
+	die "FATAL:$me: Perl dependancies not installed (", join(", " => @errorCode), ")\n";
+}
+
+# Multigraph cabability is required for this plugin
+if ( ! $ENV{'MUNIN_CAP_MULTIGRAPH'} ) {
```

Comment:
By definition the `munin-node` component is responsible for detecting plugins using `multigraph` (i.e. they are not offered to a client that lacks this capability). Thus you can omit this check here.

Reply:
Cool, didn't know that - I'll drop this check.

Reply:
@sumpfralle : sorry I didn't catch this before. Please see https://github.com/munin-monitoring/munin/blob/master/plugins/node.d/snmp__if_multi#L140 and https://github.com/munin-monitoring/munin/blob/master/lib/Munin/Plugin.pm#L613.

On the NODE side: If the plugin is run on a node that understands multigraph then MUNIN_CAP_MULTIGRAPH is set and it's safe for the plugin to run. If the node is too old the environment variable is not set and it's not safe to run the plugin.

On the MASTER side the master signals capabilities with "cap multigraph". If it sends this it will see the multigraph plugins.   If it does not it will not see the multigraph plugins. The node knows to censor the list.

But only a recent enough node.  The whole thing breaks down if the node is too old to know to detect multigraph plugins and hide them from inept masters. Therefore the test is right.  But if something more extensive like what is found in Munin::Plugin is used it also handles autoconfig right and gives a helpful (but empty) graph on the server side.

Reply:
Good to know! Thank you for correcting me.